api/handler.go

```diff
@@ -151,8 +151,9 @@ func streamResults(results chan *events.Event, w http.ResponseWriter, r *http.Re
 	for {
 		select {
 		case <-cn.CloseNotify():
-			return nil
-		case ev := <-results:
+			glog.V(3).Infof("Received CloseNotify event")
+			m.StopWatch(eventChannel.GetWatchID())
```

Comment:
In that case should I also get rid of the EventManager's cleanup thread?
It might be difficult to alert the EventManager's cleanup thread that this particular watcher is no longer needed if we only tell the EventChannel to stop. But I could change it back to the way it was originally written, so that I tell the EventChannel to stop and then, rather than putting the CleanupWatcher in a goroutine and having it always run, just call it on a case by case basis here to search for and remove any stopped EventChannel objects.